src/App.tsx

```diff
@@ -14,11 +15,108 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[],
+  selectedGoods: string[],
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    selectedGoods: ['Jam'],
+  };
+
+  updateGoods = (good: string) => {
+    this.setState((prevState: State) => {
+      return prevState.selectedGoods.includes(good)
+        ? { selectedGoods: prevState.selectedGoods.filter(item => item !== good) }
+        : { selectedGoods: [...prevState.selectedGoods, good] };
+    });
+  };
+
+  clearSelection = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  getSelectedItems = () => {
+    let selectedItems = '';
+    const { selectedGoods } = this.state;
+
+    switch (selectedGoods.length) {
+      case 0:
+        selectedItems = 'No goods selected';
+        break;
+
+      case 1:
+        selectedItems = `${selectedGoods[0]} is selected`;
+        break;
+
+      default:
+        selectedItems += selectedGoods[0];
+        for (let i = 1; i < selectedGoods.length - 1; i += 1) {
+          selectedItems += `, ${selectedGoods[i]}`;
+        }
+
+        selectedItems += ` and ${selectedGoods[selectedGoods.length - 1]} are selected`;
+    }
+
+    return selectedItems;
+  };
+
+  render() {
+    const { goods, selectedGoods } = this.state;
+    const isSelected = !!selectedGoods.length;
+    const listOfSelected = this.getSelectedItems();
+
+    return (
+      <div className="App">
+        <div className="App__content">
+          <div className="App__titles">
+            <h1 className="App__title">
+              {listOfSelected}
+            </h1>
+
+            {isSelected && (
+              <button
+                type="button"
+                onClick={this.clearSelection}
+                className="App__clear"
+              >
+                x
+              </button>
+            )}
+          </div>
+
+          <ul className="App__list">
+            {
+              goods.map((good) => (
+                <>
+                  <li
+                    key={goods.indexOf(good)}
```

Comment:
if you use fragment inside map you need to add key to fragment, not to list item